packages/react-native-oursky/src/VerifyOTP.js

```diff
@@ -0,0 +1,213 @@
+  textInputRef: RefObject<TextInput>,
+};
+
+class VerifyOTP extends React.PureComponent<Props & TextInputRefProps, State> {
+  state = {
+    value: "",
+    countDownSecond: 59,
+  };
+
+  timerId: IntervalID | null = null;
+
+  componentDidMount() {
+    this.countDown();
+  }
+
+  componentWillUnmount() {
+    if (this.timerId) {
+      clearInterval(this.timerId);
+    }
+  }
+
+  countDown = () => {
+    this.setState({ countDownSecond: 59 });
+    this.timerId = setInterval(() => {
+      this.setState(
+        prevState => ({
+          ...prevState,
+          countDownSecond: prevState.countDownSecond - 1,
+        }),
+        () => {
+          if (this.state.countDownSecond === 0 && this.timerId) {
+            clearInterval(this.timerId);
+            this.timerId = null;
+          }
+        }
+      );
+    }, 1000);
+  };
+
+  clearCode = () => {
+    this.setState({ value: "" });
+  };
+
+  onChangeText = (value: string) => {
+    if (/^\d*$/.test(value)) {
+      this.setState({ value }, () => {
+        if (this.props.onEnterCode) {
+          this.props.onEnterCode(value, this.clearCode);
+        }
+      });
+    }
+  };
+
+  onPressResend = () => {
+    if (this.props.onPressResend) {
+      this.props.onPressResend(this.countDown);
+    }
+  };
+
+  renderCodeBox = () => {
+    const { codeBoxStyle, codeBoxTextStyle } = this.props;
+    return (
+      <View style={defaultStyles.codeBoxContainer}>
+        {Array(4)
+          .fill("")
+          .map((_, idx) => {
+            return (
+              <CodeBox
+                key={idx}
+                style={codeBoxStyle}
+                textStyle={codeBoxTextStyle}
+                value={this.state.value.charAt(idx)}
+                isError={!!this.props.error}
+              />
+            );
+          })}
+      </View>
+    );
+  };
+
+  render() {
+    const {
+      description,
+      resendText,
+      resending,
+      error,
+
+      style,
+      descriptionStyle,
+      resendContainerStyle,
+      resendTextStyle,
+      errorStyle,
+
+      onEnterCode,
+
+      textInputRef,
+    } = this.props;
+
+    const { value, countDownSecond } = this.state;
+    return (
+      <View style={[defaultStyles.box, style]}>
+        <Text style={[defaultStyles.description, descriptionStyle]}>
+          {description}
+        </Text>
+        {this.renderCodeBox()}
+        <TextInput
+          ref={textInputRef}
+          value={value}
+          onChangeText={this.onChangeText}
+          autoFocus={true}
+          style={defaultStyles.hiddenTextInput}
+          keyboardType="numeric"
+          maxLength={4}
+        />
+        <TouchableOpacity
+          disabled={resending || countDownSecond !== 0}
+          style={[defaultStyles.resendContainer, resendContainerStyle]}
+          onPress={this.onPressResend}
+        >
+          <Text style={[defaultStyles.resendText, resendTextStyle]}>
+            {resendText} ({countDownSecond}
+            s)
```

Comment:
Why `)`???

Reply:
it's string, will display like this `(30s)`

Reply:
I will change this expression to make it more clear